### PR TITLE
feat: added UI for creating charges

### LINKED ISSUE #300: Implement charge UI

### PR DESCRIPTION
**Pull request by callmekungfu**
### Related Issue(s)

resolves #300

### Quick overview of the change

CleanShot 2022-10-21 at 00 15 44@2x

https://user-images.githubusercontent.com/11084692/197110376-3971330d-b9fc-4675-ac94-502d40b8e17f.mov

- Added UI for creating charges
- Added UI elements for AE after charges have been created

### CODE REVIEW

File: client/src/components/app-bar/index.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ export default function ChronicleAppBar({
             </IconButton>
           )}
           <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
-            Pemberley Springs
+            {import.meta.env.VITE_SITE_TITLE}
```
Comment (Wuon) [inline client/src/components/app-bar/index.tsx:63]: nice

File: client/src/data/charge.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export interface ChargeCreateResult {
+  id: string;
+  amount: number;
+  description: string;
+  createdDt: string;
```
Comment (Wuon) [inline client/src/data/charge.ts:5]: createdAt?

File: client/src/pages/activity-entry-details/ActivityEntryChargeDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                  label="Tip"
+                  name="tipAmount"
+                  rules={{
+                    min: 0,
+                    pattern: priceCheck,
+                  }}
+                />
+                <Spacer size="sm" />
+                <ButtonGroup size="large" fullWidth>
+                  {[0.15, 0.2, 0.25].map((p) => (
```
Comment (Wuon) [inline client/src/pages/activity-entry-details/ActivityEntryChargeDialog.tsx:64]: might be worth to have this in a constants or in a config?

File: client/src/types/activity-entry.ts [hunk 4]
```diff
@@ -29,6 +29,13 @@ interface Customer {
   updatedAt: string;
 }
 
+interface Charge {
+  id: string;
+  description: string;
+  amount: number;
+  createdDt: string;
```
Comment (Wuon) [inline client/src/types/activity-entry.ts:36]: createdAt

**Review by Wuon — APPROVED** [review]
minor nitpicks, lgtm otherwise

feel free to merge in and address the issue in a later PR

**Comment by Wuon** [comment]
Also want to ask about the user flow, but would the associate show this screen to the user for them to enter the tip amount? or is it something they would add in after they pay on their own time?

**Comment by callmekungfu** [comment]
@Wuon This would depend on what the client prefers, I sorta picked a middle of the road option for now but we can change it depending on how they feel

**Pull request merged**